### PR TITLE
Fix pipeline build publish location

### PR DESCRIPTION
**Pull request by mjbond-msft**
Related to PR #2743.  Prior to the Xamarin.Android (release) build moving from a freestyle to a pipeline build, the publish location included both the job name (xamarin-android) *and* the checkout directory name (xamarin-android) in the publish url path such as the following - note the redundant xamarin-android names (JOB_NAME/DIR_NAME) at the end:
http://xamjenkinsartifact.blob.core.windows.net/mono-jenkins-eval/xamarin-android/xamarin-android

After moving to the pipeline build based on the build.groovy jenkinsfile, the publish location changed to the following:
http://xamjenkinsartifact.blob.core.windows.net/mono-jenkins-eval/xamarin-android

This breaks anyone or any system that is looking for published packages in the original location.

The problem was introduced when the publish operation was inadvertently changed to execute from within the xamarin-android checkout directory.  Consequently, the directory itself was not included in the publish path.

The fix is to blank out the directory specifier and update the list of relative publish paths such that the publish command runs from the Jenkins workspace root directory, which is the parent of the xamarin-android checkout directory.  Consequently, the xamarin-android checkout directory now gets included in the publish link url.

### CODE REVIEW

File: build-tools/automation/build.groovy [hunk 1]
```diff
@@ -131,11 +131,15 @@ timestamps {
...
 
-        stageWithTimeout('publish test error logs to Azure', 10, 'MINUTES', XADir, false) {  // Typically takes less than a minute
+        stageWithTimeout('publish test error logs to Azure', 10, 'MINUTES', '', false) {  // Typically takes less than a minute
             echo "packaging test error logs"
-            sh "make package-test-errors"
 
-            def publishTestFilePaths = "xa-test-errors*"
+            sh """
+                cd ${XADir}
+                make package-test-errors
```
Comment (jonathanpeppers) [inline build-tools/automation/build.groovy]: could this also be `sh "make -C ${XADir} -k package-test-errors"`?

Comment (mjbond-msft) [reply build-tools/automation/build.groovy]: indeed that should work. testing the change now. commit https://github.com/xamarin/xamarin-android/pull/2766/commits/d85f005712532a916803db9418cae7f1cf08f7dd

**Comment by jonathanpeppers** [comment]
@mjbond-msft I tried @akoeplinger's idea and started a build: https://jenkins.mono-project.com/view/Xamarin.Android/job/xamarin-android/318/
```
StorageVirtualPath=xamarin-android/${env.JOB_NAME}
```
It was just set to `xamarin-android` previously.

**Comment by akoeplinger** [comment]
@jonathanpeppers looks like the env var isn't expanded there: https://jenkins.mono-project.com/view/Xamarin.Android/job/xamarin-android/318/Azure/ so you'll need to hardcode it as `xamarin-android/xamarin-android` there, though I admit I don't understand why this needs to be configurable :)

**Comment by jonathanpeppers** [comment]
@mjbond-msft it looks like this change is putting the build number in the URL: https://xamjenkinsartifact.blob.core.windows.net/mono-jenkins/xamarin-android-pr-builder-release-924/xamarin-android/bin/BuildRelease/Xamarin.Android.Sdk-OSS-9.2.99.27_HEAD_7df32b5.vsix

**Comment by jonathanpeppers** [comment]
I changed the build to use:
```
StorageVirtualPath=xamarin-android/xamarin-android
```
Although, I think this PR might be better if we get it right, because we would have to remember to do things like this on other build configurations:
```
StorageVirtualPath=xamarin-android-debug/xamarin-android
```

**Comment by jonathanpeppers** [comment]
Ok, PR builds aren't using `Jenkinsfile` yet. Example build here: https://jenkins.mono-project.com/view/Xamarin.Android/job/xamarin-android-pipe-eval/9/

URLs look right:
```
Uploaded to file storage with uri http://xamjenkinsartifact.blob.core.windows.net/mono-jenkins-eval/xamarin-android/xamarin-android/xa-build-status-v9.2.99.25_Darwin-x86_64_build-publish-fix_d85f0057-Release.zip in 00:00:07.854 (HH:mm:ss.S)
Uploaded to file storage with uri http://xamjenkinsartifact.blob.core.windows.net/mono-jenkins-eval/xamarin-android/xamarin-android/bin/BuildRelease/Xamarin.Android.Sdk-OSS-9.2.99.25_HEAD_d85f005.vsix in 00:00:27.484 (HH:mm:ss.S)
Uploaded to file storage with uri http://xamjenkinsartifact.blob.core.windows.net/mono-jenkins-eval/xamarin-android/xamarin-android/bin/Release/bundle-v21-h1b15cd2e-Release-Darwin-libzip=b95cf3f,llvm=349752c,mono=b041669.zip in 00:00:35.156 (HH:mm:ss.S)
Uploaded to file storage with uri http://xamjenkinsartifact.blob.core.windows.net/mono-jenkins-eval/xamarin-android/xamarin-android/xamarin.android-oss_v9.2.99.25_Darwin-x86_64_build-publish-fix_d85f0057-Release.zip in 00:00:46.905 (HH:mm:ss.S)
```

File: build-tools/automation/build.groovy (hunk 1 above)
Comment (mjbond-msft) [reply build-tools/automation/build.groovy]: build step executed as expected with test build `https://jenkins.mono-project.com/view/Xamarin.Android/job/xamarin-android-pipe-eval/9/`

**Pull request merged**